### PR TITLE
Webc 1688 presets refactor

### PR DESCRIPTION
**Pull request by ainvoner**

### CODE REVIEW

File: packages/ui/src/preset-item.tsx [hunk 1]
```diff
@@ -79,41 +45,45 @@ export class PresetItem {
...
-        case RelativeToTypes.After:
-          result['afterComponent'] = relativeTo.name;
-          break;
-        case RelativeToTypes.Before:
-          result['beforeComponent'] = relativeTo.name;
-          break;
-        case RelativeToTypes.Replace:
-          result['replaceComponent'] = relativeTo.name;
-          break;
+      if (!presetContainer) {
```
Comment (esakal) [inline packages/ui/src/preset-item.tsx]: I'm not sure if we need to warn about this one, as it is only relevant if someone passed an empty value. anyway the message should be self explained. something like `Cannot register component to core player using the preset manager for preset ${presetType}. preset area name is missing. Ignoring this request.`

File: packages/ui/src/preset-item.tsx [hunk 2]
```diff
@@ -79,41 +45,45 @@ export class PresetItem {
...
+  get playerConfig(): KalturaPlayerPresetComponent[] {
+    const configs: KalturaPlayerPresetComponent[] = [];
 
-    if (!containerName) {
-      this._logger.warn(`unknown container requested`, {
-        method: 'playerConfig',
-      });
-      return null;
-    }
+    for (const presetType in this._options.data.presetAreas) {
```
Comment (esakal) [inline packages/ui/src/preset-item.tsx]: presetType > presetName?

File: packages/ui/src/presets-utils.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      managerName,
+      config,
+      defaults
+    );
+    const result = {};
+    acceptableTypes.forEach(presetType => (result[presetType] = {}));
+    Object.keys(presetAreasMapping).forEach(presetName => {
+      Object.keys(presetAreasMapping[presetName]).forEach(presetType => {
+        if (acceptableTypes.indexOf(presetType) === -1) {
+          // todo [sa] add warning
```
Comment (esakal) [inline packages/ui/src/presets-utils.ts]: TBD

File: packages/ui/src/preset-item.tsx (hunk 1 above)
Comment (ainvoner) [reply packages/ui/src/preset-item.tsx]: done

File: packages/ui/src/preset-item.tsx (hunk 2 above)
Comment (ainvoner) [reply packages/ui/src/preset-item.tsx]: the current naming convention is:
presetName: Live, Playback...
presetType: playerArea, presetArea...
container: the actual preset element container name (currently similar to presetType)

File: packages/ui/src/banner-manager.tsx [hunk 4]
```diff
@@ -74,18 +74,16 @@ export class BannerManager {
...
-      options.corePlayer.config &&
-      options.corePlayer.config.contrib &&
-      options.corePlayer.config.contrib.ui &&
-      options.corePlayer.config.contrib.ui.kitchenSink
-        ? options.corePlayer.config.contrib.ui.kitchenSink
-        : {};
+    const managerConfig = ObjectUtils.get(
+      this._options.corePlayer,
+      'config.contrib.ui.banner',
+      DefaultBannerConfig
```
Comment (esakal) [inline packages/ui/src/banner-manager.tsx]: the default here is empty object, the actual default is assigned later using `mergeDefaults`

**Pull request merged**